Desktop/modules/dynamicmodules.cpp

```diff
@@ -223,9 +227,24 @@ void DynamicModules::registerForInstallingModPkg(const std::string & moduleName)
 
 void DynamicModules::stopAndRestartEngines()
 {
-	stopEngines();
+	emit stopEngines();
 	_modulesToBeUnloaded.clear(); //if we are going to restart the engines we can also forget anything that's loaded and needs to be unloaded
-	restartEngines();
+	emit restartEngines();
+}
+
+QStringList DynamicModules::importPaths() const
+{
+	QStringList allImportPaths;
+
+	for(const auto & nameMod : _modules)
+	{
+		QDir moduleFolder(tq(nameMod.second->moduleInstFolder()));
+
+		if(moduleFolder.exists("qmldir")) //So the inst contained a qmldir, and we give the library as an importpath
+			allImportPaths.append(nameMod.second->moduleRLibrary());
```

Comment:
the name moduleRLibrary is a bit awkward for the absolute path of the module folder.

Reply:
Well, it isn't the absolute path of the module folder, it is the path to the r-library that contains the installed package that is the module folder. And it is particular for the module in question.